internal/pkg/workspace/workspace_test.go

```diff
@@ -520,63 +520,135 @@ func TestWorkspace_ListPipelines(t *testing.T) {
+				prodPipelineManifest, _ := fs.Create("/copilot/pipelines/prod/manifest.yml")
+				defer prodPipelineManifest.Close()
+				prodPipelineManifest.Write([]byte(`
+name: prodManifest
 version: 1
 `))
 
-				otherInPipelinesDirBuildspec, _ := fs.Create("/copilot/pipelines/other.buildspec.yml")
-				defer otherInPipelinesDirBuildspec.Close()
-				otherInPipelinesDirBuildspec.Write([]byte(`
-name: buildspecInPipelinesDir
+				return fs
+			},
+			wantedPipelines: []PipelineManifest{
+				{
+					Name: "betaManifest",
+					Path: "/copilot/pipelines/beta/manifest.yml",
+				},
+				{
+					Name: "legacyInCopiDir",
+					Path: "/copilot/pipeline.yml",
+				},
+				{
+					Name: "prodManifest",
+					Path: "/copilot/pipelines/prod/manifest.yml",
+				},
+			},
+
+			wantedErr: nil,
+		},
+		"success finding legacy pipeline if it is the only pipeline": {
+			copilotDir: "/copilot",
+			fs: func() afero.Fs {
+				fs := afero.NewMemMapFs()
+
+				fs.Mkdir("/copilot", 0755)
+				fs.Create("/copilot/buildspec.yml")
+				legacyInCopiDirManifest, _ := fs.Create("/copilot/pipeline.yml")
+				defer legacyInCopiDirManifest.Close()
+				legacyInCopiDirManifest.Write([]byte(`
+name: legacyInCopiDir
+version: 1
 `))
 
 				return fs
 			},
-
 			wantedPipelines: []PipelineManifest{
 				{
 					Name: "legacyInCopiDir",
 					Path: "/copilot/pipeline.yml",
 				},
+			},
+			wantedErr: nil,
+		},
+		"success finding pipelines without any legacy pipelines": {
+			copilotDir: "/copilot",
+			fs: func() afero.Fs {
+				fs := afero.NewMemMapFs()
+
+				fs.Mkdir("/copilot", 0755)
+				fs.Mkdir("/copilot/pipelines", 0755)
+
+				fs.Create("/copilot/pipelines/beta/buildspec.yml")
+				betaPipelineManifest, _ := fs.Create("/copilot/pipelines/beta/manifest.yml")
+				defer betaPipelineManifest.Close()
+				betaPipelineManifest.Write([]byte(`
+name: betaManifest
+version: 1
+`))
+
+				fs.Create("/copilot/pipelines/prod/buildspec.yml")
+				prodPipelineManifest, _ := fs.Create("/copilot/pipelines/prod/manifest.yml")
+				defer prodPipelineManifest.Close()
+				prodPipelineManifest.Write([]byte(`
+name: prodManifest
+version: 1
+`))
+
+				return fs
+			},
+			wantedPipelines: []PipelineManifest{
 				{
-					Name: "otherInPipelinesDir",
-					Path: "/copilot/pipelines/other.yml",
+					Name: "betaManifest",
+					Path: "/copilot/pipelines/beta/manifest.yml",
 				},
 				{
-					Name: "legacyInPipelinesDir",
-					Path: "/copilot/pipelines/pipeline.yml",
+					Name: "prodManifest",
+					Path: "/copilot/pipelines/prod/manifest.yml",
 				},
 			},
-
 			wantedErr: nil,
 		},
+		"ignores missing manifest files": {
+			copilotDir: "/copilot",
+			fs: func() afero.Fs {
+				fs := afero.NewMemMapFs()
+
+				fs.Mkdir("/copilot", 0755)
+				fs.Mkdir("/copilot/pipelines", 0755)
+				fs.Mkdir("/copilot/pipelines/beta", 0755)
+				fs.Mkdir("/copilot/pipelines/prod", 0755)
+
+				return fs
+			},
+			wantedPipelines: nil,
+			wantedErr:       nil,
+		},
```

Comment:
Totally my bad, but can we add some unit tests for catching errs?

Reply:
ok I added some tests to cover as many errors as possible, not sure if they are over the top or not :) let me know if you want to keep all of the new tests or not.

Reply:
Love the tests. Thank you!